shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/module/HomeViewModel.kt

```diff
@@ -30,16 +35,40 @@ class HomeViewModel(
     private var currentPage = 1
 
     fun fetchData() {
-        getSurveysUseCase(currentPage, 10, isForceLatestData = false)
+        getProfile()
             .onStart { setStateLoading() }
-            .catch { emit(listOf()) }
+            .combine(fetchSurvey(1, 10, true)) { user, surveys ->
+                Pair(user, surveys)
+            }
             .onEach {
-                currentPage++
                 handleFetchSuccess()
             }
             .launchIn(viewModelScope)
     }
 
+    private fun getProfile(): Flow<User?> {
+        return flow {
+            getUserProfileUseCase()
+                .catch { emit(null) }
+                .collect { emit(it) }
+        }
+    }
+
+    private fun fetchSurvey(
+        page: Int,
+        pageSize: Int,
+        isForceLatestData: Boolean
+    ): Flow<List<Survey>> {
+        return flow {
+            getSurveysUseCase(page, pageSize, isForceLatestData = false)
+                .catch { emit(listOf()) }
+                .collect {
+                    currentPage = page+1
+                    emit(it)
+                }
+        }
```

Comment:
We can do this instead.
```suggestion
        return getSurveysUseCase(page, pageSize, isForceLatestData = isForceLatestData)
            .catch { emit(listOf()) }
            .onEach {
                currentPage = page + 1
            }
```